src/Moq/ExpressionExtensions.cs

```diff
@@ -421,5 +421,10 @@ public static string ToStringFixed(this Expression expression)
 		{
 			return new StringBuilder().AppendExpression(expression).ToString();
 		}
+
+		public static object[] ExtractConstructorCallArguments<T>(this Expression<Func<T>> constructorExpression)
+		{
+			return ConstructorCallVisitor.ExtractArgumentValues(constructorExpression);
+		}
```

Comment:
If this method remains a one-liner, I think you could just as well inline it everywhere.

However, it might be better to keep it, but add some basic validation:

* Throw `ArgumentNullException` for `constructorExpression` if it is `null`.
* Throw `ArgumentException` when `constructorExpression`'s `.Body` does not actually represent a `new` expression.

Optionally document that these exceptions may get thrown in the new `new Mock<T>` / `MockRepository.Create` overloads.

Reply:
Finally inlined it... first exception was missing, second exception when `.Body` doesn't represent a `new` expression was already managed.